File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/ApplicationExceptions.java
// VeriBlock PoP Miner
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.

package nodecore.miners.pop.contracts;

public class ApplicationExceptions {
    public static class ExceededMaxTransactionFee extends RuntimeException {}

    public static class PoPSubmitRejected extends RuntimeException {}

    public static class CorruptSPVChain extends RuntimeException {
        public CorruptSPVChain(String message) {
            super(message);
        }
    }

    public static class UnableToAcquireTransactionLock extends RuntimeException {}

    public static class DuplicateTransactionException extends RuntimeException {}

    public static class SendTransactionException extends RuntimeException {
        public SendTransactionException (Exception inner) {
            this.addSuppressed(inner);
        }
    }
}

File: nodecore-miners-pop/src/main/java/nodecore/miners/pop/contracts/ApplicationExceptions.kt
// VeriBlock PoP Miner
// Copyright 2017-2019 Xenios SEZC
// All rights reserved.
// https://www.veriblock.org
// Distributed under the MIT software license, see the accompanying
// file LICENSE or http://www.opensource.org/licenses/mit-license.php.
package nodecore.miners.pop.contracts

class ApplicationExceptions {
    class ExceededMaxTransactionFee : RuntimeException()
    class PoPSubmitRejected : RuntimeException()
    class CorruptSPVChain(message: String?) : RuntimeException(message)
    class UnableToAcquireTransactionLock : RuntimeException()
    class DuplicateTransactionException : RuntimeException()
    class SendTransactionException(inner: Exception?) : RuntimeException() {
        init {
            this.addSuppressed(inner)
        }
    }
}
